api/backend.go

```diff
@@ -555,6 +558,53 @@ func (b *StatusBackend) GetNodesFromContract(rpcEndpoint string, contractAddress
 	return response, nil
 }
 
+// VerifyENSName verifies that a registered ENS name matches the expected public key
+func (b *StatusBackend) VerifyENSName(ensName, publicKeyStr, rpcEndpoint, contractAddress string) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), contractQueryTimeout)
+	defer cancel()
+
+	pubKeyBytes, err := hex.DecodeString(publicKeyStr)
+	if err != nil {
+		return false, err
+	}
+
+	expectedPubKey, err := btcec.ParsePubKey(pubKeyBytes, btcec.S256())
+	if err != nil {
+		return false, err
+	}
+
+	ethclient, err := ethclient.DialContext(ctx, rpcEndpoint)
+	if err != nil {
+		return false, err
+	}
+
+	// Resolve ensName
+	resolver, err := ens.NewResolver(ethclient, ensName)
+	if err != nil {
+		b.log.Error("error while creating ENS name resolver", "ensName", ensName, "error", err)
+		return false, err
+	}
+	x, y, err := resolver.PubKey()
+	if err != nil {
+		b.log.Error("error while resolving public key from ENS name", "ensName", ensName, "error", err)
+		return false, err
+	}
+
+	// Assemble the bytes returned for the pubkey
+	pb := make([]byte, btcec.PubKeyBytesLenUncompressed)
```

Comment:
I would try to stick to `github.com/ethereum/go-ethereum/crypto`. Which means doing something like this:
```go
// crypto/elliptic
// github.com/ethereum/go-ethereum/crypto
pubKeyBytes := elliptic.Marshal(crypto.S256(), new(big.Int).SetBytes(x), new(big.Int).SetBytes(y))
return bytes.Equal(expectedPubKeyBytes, pubKeyBytes)
```